WPT export is broken

https://github.com/servo/servo/actions/runs/7180806387/job/19553860946

```
INFO:root:  → Execution (cwd='./wpt'): git checkout -b servo_export_30767
INFO:root:  → Execution (cwd='./wpt'): git fetch --unshallow origin
INFO:root:  → Execution (cwd='./wpt'): git push -f ***github.com/servo-wpt-sync/web-platform-tests.git servo_export_30767
INFO:root:  → Execution (cwd='./wpt'): git checkout main
WARNING:root:Process execution failed with output:
error: pathspec 'main' did not match any file(s) known to git

INFO:root:  → Request: POST repos/web-platform-tests/wpt/pulls
INFO:root:  → Request JSON: {'title': 'Add initial support for offscreen rendering', 'head': 'servo-wpt-sync:servo_export_30767', 'base': 'main', 'body': 'This patch makes it possible to direct the output of the compositor to an offscreen OpenGL framebuffer object, rather than the OpenGL backbuffer or an image in IpcSharedMemory or a png file.\r\n\r\nWe also make servoshell use this iff the minibrowser is enabled, and rework CompositeTarget to allow the embedder to specify the target and output file if any when initialising Servo.\r\n\r\nCompositeTarget before:\r\n\r\n- Window draws to the OpenGL backbuffer only\r\n- WindowAndPng draws to an uncompressed image in shared memory only\r\n    - the name was misleading; it’s neither WindowAnd nor Png\r\n- PngFile draws to IOCompositor.output_file only (which must be Some) and exits\r\n\r\nCompositeTarget after:\r\n\r\n- Window draws to the OpenGL backbuffer only\r\n- Fbo draws to an offscreen OpenGL framebuffer object (IOCompositor::output_framebuffer_id)\r\n- SharedMemory draws to an uncompressed image in shared memory only\r\n- PngFile(String) draws to a PNG file at the given path only and exits\r\n\r\nManually verified working on:\r\n- [x] Windows\r\n- [x] macOS\r\n- [x] Linux (X11)\r\n- [x] Linux (Wayland) — no worse than before\r\n    - #<!-- nolink -->30826\r\n    - #<!-- nolink -->30827\r\n\r\nReviewed in servo/servo#30767', 'maintainer_can_modify': False}
[...]
ERROR:root:Got unexpected 422 response: {"message":"Validation Failed","errors":[{"resource":"PullRequest","field":"base","code":"invalid"}],"documentation_url":"https://docs.github.com/rest/pulls/pulls#create-a-pull-request"}
Traceback (most recent call last):
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/__init__.py", line 195, in run
    run.run()
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/__init__.py", line 106, in run
    step.run(self)
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/step.py", line 292, in run
    pull_request = self.target_repo.open_pull_request(
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/github.py", line 112, in open_pull_request
    response = authenticated(self.sync, "POST", self.pulls_url, json=data)
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/github.py", line 47, in authenticated
    raise ValueError(
ValueError: Got unexpected 422 response: {"message":"Validation Failed","errors":[{"resource":"PullRequest","field":"base","code":"invalid"}],"documentation_url":"https://docs.github.com/rest/pulls/pulls#create-a-pull-request"}
Error: Process completed with exit code 1.
```

--- a/python/wpt/exporter/__init__.py
+++ b/python/wpt/exporter/__init__.py
@@ -19,6 +19,7 @@ import dataclasses
 import json
 import logging
 import re
+import shutil
 import subprocess
 
 from typing import Callable, Optional
@@ -50,8 +51,12 @@ class LocalGitRepo:
         self.path = path
         self.sync = sync
 
+        # We pass env to subprocess, which may clobber PATH, so we need to find
+        # git in advance and run the subprocess by its absolute path.
+        self.git_path = shutil.which("git")
+
     def run_without_encoding(self, *args, env: dict = {}):
-        command_line = ["git"] + list(args)
+        command_line = [self.git_path] + list(args)
         logging.info("  → Execution (cwd='%s'): %s",
                      self.path, " ".join(command_line))
 

--- a/python/wpt/exporter/__init__.py
+++ b/python/wpt/exporter/__init__.py
@@ -142,9 +142,9 @@ class WPTSync:
     suppress_force_push: bool = False
 
     def __post_init__(self):
-        self.servo = GithubRepository(self, self.servo_repo)
-        self.wpt = GithubRepository(self, self.wpt_repo)
-        self.downstream_wpt = GithubRepository(self, self.downstream_wpt_repo)
+        self.servo = GithubRepository(self, self.servo_repo, "main")
+        self.wpt = GithubRepository(self, self.wpt_repo, "master")
+        self.downstream_wpt = GithubRepository(self, self.downstream_wpt_repo, "master")
         self.local_servo_repo = LocalGitRepo(self.servo_path, self)
         self.local_wpt_repo = LocalGitRepo(self.wpt_path, self)
 

--- a/python/wpt/exporter/github.py
+++ b/python/wpt/exporter/github.py
@@ -55,10 +55,10 @@ class GithubRepository:
     This class allows interacting with a single GitHub repository.
     """
 
-    def __init__(self, sync: WPTSync, repo: str, main_branch_name: str):
+    def __init__(self, sync: WPTSync, repo: str, default_branch: str):
         self.sync = sync
         self.repo = repo
-        self.main_branch_name = main_branch_name
+        self.default_branch = default_branch
         self.org = repo.split("/")[0]
         self.pulls_url = f"repos/{self.repo}/pulls"
 
@@ -106,7 +106,7 @@ class GithubRepository:
         data = {
             "title": title,
             "head": branch.get_pr_head_reference_for_repo(self),
-            "base": self.main_branch_name,
+            "base": self.default_branch,
             "body": body,
             "maintainer_can_modify": False,
         }

--- a/python/wpt/exporter/github.py
+++ b/python/wpt/exporter/github.py
@@ -55,9 +55,10 @@ class GithubRepository:
     This class allows interacting with a single GitHub repository.
     """
 
-    def __init__(self, sync: WPTSync, repo: str):
+    def __init__(self, sync: WPTSync, repo: str, main_branch_name: str):
         self.sync = sync
         self.repo = repo
+        self.main_branch_name = main_branch_name
         self.org = repo.split("/")[0]
         self.pulls_url = f"repos/{self.repo}/pulls"
 
@@ -74,7 +75,7 @@ class GithubRepository:
         self, branch: GithubBranch
     ) -> Optional[PullRequest]:
         """If this repository has an open pull request with the
-        given source head reference targeting the master branch,
+        given source head reference targeting the main branch,
         return the first matching pull request, otherwise return None."""
 
         params = "+".join([
@@ -105,7 +106,7 @@ class GithubRepository:
         data = {
             "title": title,
             "head": branch.get_pr_head_reference_for_repo(self),
-            "base": "master",
+            "base": self.main_branch_name,
             "body": body,
             "maintainer_can_modify": False,
         }

--- a/python/wpt/exporter/github.py
+++ b/python/wpt/exporter/github.py
@@ -74,7 +74,7 @@ class GithubRepository:
         self, branch: GithubBranch
     ) -> Optional[PullRequest]:
         """If this repository has an open pull request with the
-        given source head reference targeting the main branch,
+        given source head reference targeting the master branch,
         return the first matching pull request, otherwise return None."""
 
         params = "+".join([
@@ -105,7 +105,7 @@ class GithubRepository:
         data = {
             "title": title,
             "head": branch.get_pr_head_reference_for_repo(self),
-            "base": "main",
+            "base": "master",
             "body": body,
             "maintainer_can_modify": False,
         }

--- a/python/wpt/exporter/step.py
+++ b/python/wpt/exporter/step.py
@@ -190,7 +190,7 @@ class CreateOrUpdateBranchForPRStep(Step):
             return branch_name
         finally:
             try:
-                run.sync.local_wpt_repo.run("checkout", "main")
+                run.sync.local_wpt_repo.run("checkout", "master")
                 run.sync.local_wpt_repo.run("branch", "-D", branch_name)
             except Exception:
                 pass

--- a/python/wpt/test.py
+++ b/python/wpt/test.py
@@ -644,17 +644,17 @@ def setUpModule():
         suppress_force_push=True,
     )
 
-    def setup_mock_repo(repo_name, local_repo):
+    def setup_mock_repo(repo_name, local_repo, main_branch_name: str):
         subprocess.check_output(
             ["cp", "-R", "-p", os.path.join(TESTS_DIR, repo_name), local_repo.path])
-        local_repo.run("init", "-b", "main")
+        local_repo.run("init", "-b", main_branch_name)
         local_repo.run("add", ".")
         local_repo.run("commit", "-a", "-m", "Initial commit")
 
     logging.info("=" * 80)
     logging.info("Setting up mock repositories")
-    setup_mock_repo("servo-mock", SYNC.local_servo_repo)
-    setup_mock_repo("wpt-mock", SYNC.local_wpt_repo)
+    setup_mock_repo("servo-mock", SYNC.local_servo_repo, "main")
+    setup_mock_repo("wpt-mock", SYNC.local_wpt_repo, "master")
     logging.info("=" * 80)
 
 

--- a/python/wpt/test.py
+++ b/python/wpt/test.py
@@ -644,10 +644,10 @@ def setUpModule():
         suppress_force_push=True,
     )
 
-    def setup_mock_repo(repo_name, local_repo, main_branch_name: str):
+    def setup_mock_repo(repo_name, local_repo, default_branch: str):
         subprocess.check_output(
             ["cp", "-R", "-p", os.path.join(TESTS_DIR, repo_name), local_repo.path])
-        local_repo.run("init", "-b", main_branch_name)
+        local_repo.run("init", "-b", default_branch)
         local_repo.run("add", ".")
         local_repo.run("commit", "-a", "-m", "Initial commit")
 

